### PR TITLE
#437 : Roles and permission in module

### PR DESCRIPTION
**Pull request by Fhadk**
Please review changes

### CODE REVIEW

File: meveo-model/src/main/java/org/meveo/model/security/Permission.java [hunk 1]
```diff
@@ -49,13 +52,14 @@
  */
 @Entity
 @Cacheable
+@ModuleItem(path = "Permission", value = "Permissions")
```
Comment (ClementBareth) [inline meveo-model/src/main/java/org/meveo/model/security/Permission.java]: Value should be "Permission" and path "permission"

File: meveo-model/src/main/java/org/meveo/model/security/Permission.java [hunk 2]
```diff
@@ -84,6 +88,16 @@
     @JoinColumn(name = "category")
     private PermissionCategory category;
     
+    @Column(name = "code", nullable = false, length = 255)
```
Comment (ClementBareth) [inline meveo-model/src/main/java/org/meveo/model/security/Permission.java]: You don't need to add code & description as they are managed in BusinessEntity

Also, you can remove the "implementd IEntity<Long>"

**Review by ClementBareth — CHANGES_REQUESTED** [review]

**Comment by ClementBareth** [comment]
@Fhadk please review the comments

File: meveo-model/src/main/java/org/meveo/model/security/Permission.java (hunk 2 above)
Comment (Fhadk) [reply meveo-model/src/main/java/org/meveo/model/security/Permission.java]: Rectified, deleted(code and description) from both files.

File: meveo-model/src/main/java/org/meveo/model/security/Permission.java (hunk 1 above)
Comment (Fhadk) [reply meveo-model/src/main/java/org/meveo/model/security/Permission.java]: Rectified

**Review by Fhadk — COMMENTED** [review]
@ClementBareth  Please review incorporated changes.

File: meveo-model/src/main/java/org/meveo/model/security/Role.java [hunk 3]
```diff
@@ -37,6 +39,7 @@
  */
 @Entity
 @Cacheable
+@ModuleItem(path = "Role", value = "Roles")
```
Comment (ClementBareth) [inline meveo-model/src/main/java/org/meveo/model/security/Role.java]: Here, same logic apply. It should be path = "roles" & value = "Role

**Review by ClementBareth — CHANGES_REQUESTED** [review]

**Comment by ClementBareth** [comment]
@Fhadk

File: meveo-model/src/main/java/org/meveo/model/security/Role.java (hunk 3 above)
Comment (Fhadk) [reply meveo-model/src/main/java/org/meveo/model/security/Role.java]: rectified

Comment (ClementBareth) [reply meveo-model/src/main/java/org/meveo/model/security/Role.java]: @Fhadk I don't see any changes - did you commit ?

**Review by ClementBareth — APPROVED** [review]

**Pull request merged**